### PR TITLE
Syslog integration

### PR DESCRIPTION
**Pull request by aantthony**
Rewrite using a decoupled syslog and console output.

This will be version 2.0.0 since it breaks compatibility.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1,32 +1,113 @@
...
   }
 
-  var transports = transportFactory(opts);
-  return new Logger(transports, opts);
-};
+  function Logger (prefix) {
+    this.component = prefix
+  }
+
+  LEVELS
```
Comment (smithamax) [inline lib/index.js:64]: Only sub loggers get level methods? Would be nice if LOKELogger was just the logger, and then sub loggers inherited all the methods, I think.

Comment (aantthony) [inline lib/index.js:64]: Yeah, only the sub loggers do. The reason to only allow sub loggers and not a root logger is for consistency.

**Comment by aantthony** [comment]
Switched to https://github.com/sindresorhus/eslint-config-xo for linting.

**Comment by smithamax** [comment]
:+1:

**Pull request merged**